Tasks/RunDistributedTests/CheckTestAgentCompat.ps1

```diff
@@ -0,0 +1,76 @@
+function Check-AgentCompat($ProductVersion)
+{
+    try
+    {
+        $InstalledVersionRegValueName = "Version"
+        $InstalledVersionRegKey = ("SOFTWARE\Microsoft\DevDiv\vstf\Servicing\{0}\testExecCore" -f $ProductVersion)
+
+        if ($InstalledVersionRegKey -and $InstalledVersionRegValueName)
+	    {
+          $versionString = $null
+
+          #if 64bit OS, check 64bit registry view first
+          if ((Get-WmiObject -Class Win32_OperatingSystem -ComputerName "localhost" -ea 0).OSArchitecture -eq '64-bit')
+          {
+            $versionString = Get-RegistryValue -RegistryHive LocalMachine -Key $InstalledVersionRegKey -Value $InstalledVersionRegValueName -RegistryView Registry64
+          }
+          
+          if($versionString -eq $null)
+          {          
+            $versionString = Get-RegistryValue -RegistryHive LocalMachine -Key $InstalledVersionRegKey -Value $InstalledVersionRegValueName -RegistryView Registry32
+          }
+
+          if($versionString)
+          {
+            $versionArray = $versionString.split('.')            
+
+            $minorVersion=$null
+            $majorVersion=$null
+            [Int]::TryParse($versionArray[0], [ref]$majorVersion)
+            [Int]::TryParse($versionArray[2], [ref]$minorVersion)
+
+            if($majorVersion -eq 14 -and $minorVersion -le 24712)
```

Comment:
This is what we shipped last. From testexeccore registry of a freshly installed TA machine.